sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDB.scala

```diff
@@ -965,12 +969,33 @@ class RocksDB(
 
   /**
    * Function to release RocksDB instance lock that allows for synchronized access to the state
-   * store instance
+   * store instance. Optionally provide a thread to check against, and release only if provided
+   * thread is the one that acquired the lock.
    *
    * @param opType - operation type releasing the lock
+   * @param releaseForThreadOpt - optional thread to check against acquired thread
    */
-  private def release(opType: RocksDBOpType): Unit = acquireLock.synchronized {
+  private def release(
+      opType: RocksDBOpType,
+      releaseForThreadOpt: Option[AcquiredThreadInfo] = None): Unit = acquireLock.synchronized {
     if (acquiredThreadInfo != null) {
+      if (releaseForThreadOpt.nonEmpty) {
+        if (releaseForThreadOpt.get.threadRef.get.isEmpty) {
+          logInfo(log"Thread reference is empty when attempting to release for" +
+            log" opType=${MDC(LogKeys.OP_TYPE, opType.toString)}, ignoring release." +
+            log" Lock is held by ${MDC(LogKeys.THREAD, acquiredThreadInfo)}")
+          return
```

Comment:
could we just do `if else` here ? i guess `return` is not recommended for scala per style guide ?